django_setup_configuration/model_utils.py

```diff
@@ -65,14 +66,51 @@ def settings_customise_sources(
             file_secret_settings: SecretsSettingsSource,
         ) -> tuple[PydanticBaseSettingsSource, ...]:
             # Note: lower indices have higher priority
+            yaml_config_settings_source = YamlConfigSettingsSource(
+                settings_cls,
+                yaml_file=yaml_file,
+            )
+
+            yaml_data = yaml_config_settings_source.yaml_data
+
+            def substitute_secret(field: dict, field_name: str):
+                if "env" in field["valueFrom"]:
+                    field_value = field["valueFrom"]["env"]
+                    if field_value in os.environ:
+                        return os.environ[field_value]
+                    else:
+                        raise ValueError(
+                            f"Missing environment variable for field '{field_name}'."
+                        )
+                else:
+                    raise ValueError(
+                        "An environment variable ('env') must be specified "
+                        f"in dictionary field '{field_name}'."
+                    )
+
+            if namespace in yaml_data:
+                config_dict = yaml_data[namespace]
+                for field_name in config_dict.keys():
+                    field = config_dict[field_name]
+                    if isinstance(field, dict) and "valueFrom" in field:
+                        yaml_data[namespace][field_name] = substitute_secret(
+                            field, field_name
+                        )
+                    elif isinstance(field, list):
+                        objects = field
+                        for obj in objects:
+                            for obj_field_name in obj.keys():
+                                obj_field = obj[obj_field_name]
+                                if (
+                                    isinstance(obj_field, dict)
+                                    and "valueFrom" in obj_field
+                                ):
+                                    obj[obj_field_name] = substitute_secret(
+                                        obj_field, obj_field_name
+                                    )
```

Comment:
This works for configurations like this:
```yaml
some_model_enable: True
some_model:
  some_config:
    some_string:
      formValue:
        env: BLA
```
But if the secret is nested further, then it will not be picked up. For example:

```yaml
some_model_enable: True
some_model:
  some_config:
    some_dict:
      some_str:
        formValue:
          env: BLA
```
Maybe we could consider turning this into a recursive function?

I don't think there is any configuration yet that requires more than this nesting, but I feel that it could happen in the future 😅 

Also, a recursive function could also remove the need to duplicate the logic of `if isinstance(field, dict) and "valueFrom" in field` that is repeated when iterating over obj/obj_field?